lighthouse-core/test/audits/dobetterweb/has-inspector-issues-test.js

```diff
@@ -0,0 +1,123 @@
+      blockedByResponse: [],
+      /** @type {Array<LH.Crdp.Audits.HeavyAdIssueDetails>} */
+      heavyAds: [],
+      /** @type {Array<LH.Crdp.Audits.ContentSecurityPolicyIssueDetails>} */
+      contentSecurityPolicy: [],
+    };
+    const auditResult = InspectorIssuesAudit.audit({
+      InspectorIssues: issues,
+    });
+    expect(auditResult.score).toBe(1);
+    expect(auditResult.details.items).toHaveLength(0);
+  });
+
+  it('lists the correct description with the associated issue type', () => {
+    const issues = {
+      mixedContent: [
+        {
+          resolutionStatus: 'MixedContentBlocked',
+          insecureURL: 'www.mixedcontent.com',
+          mainResourceURL: 'www.mixedcontent.com',
+        },
+      ],
+      sameSiteCookies: [
+        {
+          cookieUrl: 'www.samesitecookies.com',
+        },
+      ],
+      blockedByResponse: [
+        {
+          reason: 'CoepFrameResourceNeedsCoepHeader',
+          request: {
+            url: 'www.blockedbyresponse.com',
+          },
+        },
+        {
+          reason: 'CoopSandboxedIFrameCannotNavigateToCoopPage',
+          request: {
+            url: 'www.blockedbyresponse.com',
+          },
+        },
+        {
+          reason: 'CorpNotSameOriginAfterDefaultedToSameOriginByCoep',
+          request: {
+            url: 'www.blockedbyresponse.com',
+          },
+        },
+        {
+          reason: 'CorpNotSameOrigin',
+          request: {
+            url: 'www.blockedbyresponse.com',
+          },
+        },
+        {
+          reason: 'CorpNotSameSite',
+          request: {
+            url: 'www.blockedbyresponse.com',
+          },
+        },
+      ],
+      heavyAds: [
+        {
+          reason: 'NetworkTotalLimit',
+        },
+        {
+          reason: 'CpuTotalLimit',
+        },
+        {
+          reason: 'CpuPeakLimit',
+        },
+      ],
+      contentSecurityPolicy: [
+        {
+          contentSecurityPolicyViolationType: 'kInlineViolation',
+          blockedUrl: 'www.contentsecuritypolicy.com',
+        },
+        {
+          contentSecurityPolicyViolationType: 'kEvalViolation',
+          blockedUrl: 'www.contentsecuritypolicy.com',
+        },
+        {
+          contentSecurityPolicyViolationType: 'kURLViolation',
+          blockedUrl: 'www.contentsecuritypolicy.com',
+        },
+        // These last two should be filtered out as they aren't supported yet
+        {
+          contentSecurityPolicyViolationType: 'kTrustedTypesSinkViolation',
+          blockedUrl: 'www.contentsecuritypolicy.com',
+        },
+        {
+          contentSecurityPolicyViolationType: 'kTrustedTypesPolicyViolation',
+          blockedUrl: 'www.contentsecuritypolicy.com',
+        },
+      ],
+    };
+
+    const auditResult = InspectorIssuesAudit.audit({
+      InspectorIssues: issues,
+    });
+    expect(auditResult.score).toBe(0);
+    expect(auditResult.details.items).toHaveLength(13);
```

Comment:
maybe it would be more advantageous to split this up into several tests with the assertions inline (using `toMatchObject`, for instance) rather than one giant snapshot in a separate file? It would also allow for testing a few more of the cases for each type of issue (`requestUrl`, filtered CSP types, maybe?, etc)

Reply:
any thoughts on this?

Reply:
Ah, seems I missed this comment before. After the changes proposed in the eng sync, I will likely be breaking up the tests by type anyways.